applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_local_stress_response_function.h

```diff
@@ -0,0 +1,599 @@
+
+		int stress_vec_size = element_stress.size();
+
+		if(m_stress_treatment == "mean")
+		{
+			for(int i = 0; i < stress_vec_size; i++)
+				m_stress_value += element_stress[i];
+
+			m_stress_value /= stress_vec_size;
+		}
+		else if(m_stress_treatment == "GP")
+		{
+			if(stress_vec_size >= m_id_of_location)
+				m_stress_value = element_stress[m_id_of_location - 1];
+			else
+				KRATOS_ERROR << "Chosen Gauss-Point is not availible. Chose 'stress_location' between 1 and " <<
+								stress_vec_size  << "!"<< std::endl;
+		}
+		else if(m_stress_treatment == "node")
+		{
+			const int num_ele_nodes = m_traced_pElement->GetGeometry().PointsNumber();
+			if(num_ele_nodes >= m_id_of_location)
+				m_stress_value = element_stress[m_id_of_location - 1];
+			else
+				KRATOS_ERROR << "Chosen Node is not availible. The element has only " <<
+								num_ele_nodes  << " nodes."<< std::endl;
+
+		}
+
+		// Set initial value if not done yet
+		/*if(!m_initial_value_defined)
+		{
+			m_initial_value = m_stress_value;
+			m_initial_value_defined = true;
+		}*/
+
+		return m_stress_value;
+
+		KRATOS_CATCH("");
+	}
+	// --------------------------------------------------------------------------
+	/*double GetInitialValue() old function, don't needed for sensitivity analysis
+	{
+		KRATOS_TRY;
+
+		if(!m_initial_value_defined)
+			KRATOS_THROW_ERROR(std::logi:error, "Initial value not yet defined! First compute it by calling \"CalculateValue()\"", m_initial_value_defined);
+
+		return m_initial_value;
+
+		KRATOS_CATCH("");
+	}
+
+	// --------------------------------------------------------------------------
+	double GetValue() old function, don't needed for sensitivity analysis
+	{
+		KRATOS_TRY;
+
+		return m_stress_value;
+
+		KRATOS_CATCH("");
+	}*/
+
+	// ==============================================================================
+
+	///@}
+	///@name Access
+	///@{
+
+	///@}
+	///@name Inquiry
+	///@{
+
+	///@}
+	///@name Input and output
+	///@{
+
+	/// Turn back information as a string.
+	/*virtual std::string Info() const            old function, don't needed for sensitivity analysis
+	{
+		return "AdjointLocalStressResponseFunction";
+	}
+
+	/// Print information about this object.
+	virtual void PrintInfo(std::ostream &rOStream) const    old function, don't needed for sensitivity analysis
+	{
+		rOStream << "AdjointLocalStressResponseFunction";
+	}
+
+	/// Print object's data.
+	virtual void PrintData(std::ostream &rOStream) const    old function, don't needed for sensitivity analysis
+	{
+	}*/
+
+	///@}
+	///@name Friends
+	///@{
+
+	///@}
+
+	// ==============================================================================
+	void CalculateGradient(const Element& rAdjointElem, const Matrix& rAdjointMatrix,
+                                   Vector& rResponseGradient,
+                                   ProcessInfo& rProcessInfo) override
+	{
+
+		rResponseGradient.resize(rAdjointMatrix.size1());
```

Comment:
check if the size is already correct prior to resizing (important for performance reasons)